xscope_fileio/src/xscope_io_device.c

```diff
@@ -93,6 +94,12 @@ size_t xscope_fread(xscope_file_t *xscope_file, uint8_t *buffer, size_t n_bytes_
 
     xscope_bytes(XSCOPE_ID_READ_BYTES, sizeof(packet), packet);
 
+    // Add a delay to avoid a race condition seen only on Windows
+    // See issue 30
+    #define XSCOPE_FREAD_RACE_CONDITION_DELAY ( 10000 )
+    uint32_t time_delay = get_reference_time() + XSCOPE_FREAD_RACE_CONDITION_DELAY;
```

Comment:
Yep, looks good! The only thing to make it slightly more generic is to say

#define XSCOPE_FREAD_RACE_CONDITION_DELAY (XS1_TIMER_MHZ * 100) 

but since 100MHz for the timer is generally a constant, the above is OK.

I still intend to fully understand the timer mystery.